lib/Dialect/Torch/Transforms/ReduceOpVariants.cpp

```diff
@@ -39,7 +39,8 @@ class ConvertToImmutableTensors : public RewritePattern {
         opOperand.set(rewriter.create<CopyToValueTensorOp>(op->getLoc(),
                                                            opOperand.get()));
       } else if (auto listType = operandType.dyn_cast<ListType>()) {
-        if (!listType.getContainedType().isa<NonValueTensorType>())
+        if (!(listType.getContainedType().isa<NonValueTensorType>() ||
+              listType.getContainedType().isa<OptionalType>()))
```

Comment:
Make sure to coordinate with @pashu123 on this fix, since we now have two PRs trying to fix this. See also @cathyzhyi's comment: https://github.com/llvm/torch-mlir/pull/488#pullrequestreview-834314301